### PR TITLE
[Rust] Use clap to parse REPL command line args

### LINKED ISSUE #823: oso cli help
1. from bash query  `oso --help`
2. EXPECTED: prints help message with command line options for the oso cli tool
3. OBSERVED:
```
nniepoo@anniesasus:~/oso/osopractice$ oso --help
Error: policy files must have the .polar extension. --help does not.
anniepoo@anniesasus:~/oso/osopractice$ 
```

Suggested fix - make -h -help and --help all print a sane error message

### PR DESCRIPTION
**Pull request by joshrotenberg**
This replaces `std::env::args` with [clap](https://crates.io/crates/clap) for command line parsing in the REPL. 

Fixes #823 

PR checklist:

- [ ] Added changelog entry.

### CODE REVIEW

**Comment by gj** [comment]
Will get this looked at tomorrow @joshrotenberg — thanks for the PR!

**Review by gj — APPROVED** [review]
Tested locally and LGTM.

File: languages/rust/oso/Cargo.toml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [package]
```
[the remark this answers is not in the document]
Comment (joshrotenberg) [reply languages/rust/oso/Cargo.toml:1]: You're right, though, should've been a different commit or PR.

[the remark this answers is not in the document]
Comment (gj) [reply languages/rust/oso/Cargo.toml:1]: Next time 🙂 

Thanks again for the contribution!

**Pull request merged**